### DISCUSSION TITLE
fill colors gdf

### DISCUSSION THREAD

**Discussion by rh-afk** [opengeos/leafmap#527] [Q&A] [answered]
Hi @giswqs , I just run your example geopandas notebook [https://leafmap.org/notebooks/13_geopandas/](url) , however fill colors seems not working. How to solve this issue. thanks
![image](https://github.com/opengeos/leafmap/assets/141318843/03de9867-6016-4cca-94c2-c6c280174765)

**Comment by giswqs** [comment, marked as answer]
Run `leafmap.update_package()` and restart the kernel. Alternatively, you can control the style. 

```python
countries = "https://open.gishub.org/data/world/countries.geojson"
m = leafmap.Map(center=[0, 0], zoom=2)
style = {
    "stroke": True,
    "color": "#0000ff",
    "weight": 2,
    "opacity": 1,
    "fill": True,
    "fillColor": "#0000ff",
    "fillOpacity": 0.1,
}
hover_style = {"fillOpacity": 0.7}
m.add_geojson(countries, layer_name="Countries", style=style, hover_style=hover_style)
m
```

**Reply by rh-afk** [reply]
thanks @giswqs , alternative way with hover did it. if i just update package, then restart kernel and run again I just got a map without any geojson layer there. I need to reinstall leafmap to be enable shown the layer but without fill color as previously mentioned in previous discussion

**Reply by giswqs** [reply]
v0.23.4 has been released.

**Reply by rh-afk** [reply]
thanks @giswqs fill color only with 1 color using the latest leafmap (v0.23.4), is that possible to control the multi color for gdf
or fill polygon/shape file in gdf with different color and make a legend

**Reply by rh-afk** [reply]
Hi @giswqs , I just try manual to separate shape file into several gdf and put different style . It works to showing different color, however icon of information will be repeated as many as add_gdf put there. is there any ways to combine all of that and pointing the color based on several columns in single gdf, instead of use manual one by one .
Also if possible we can put legend based on  several columns to identify the shape.

thanks
